### PR TITLE
added otp code & pass set by user attributes to user

### PR DESCRIPTION
**Pull request by osiievska23**

### CODE REVIEW

File: src/main/java/com/icthh/xm/uaa/service/dto/UserDTO.java [hunk 1]
```diff
@@ -97,6 +97,15 @@ public class UserDTO {
 
     private Instant lastLoginDate;
 
+    /**
+     * User has set password during registration himself
+     */
+    private Boolean passwordSetByUser;
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/service/dto/UserDTO.java:103]: do we need this fields in UserDTO? it's can be change only by logic, not by end-user

File: src/main/resources/config/liquibase/changelog/20241004170000_add_otp_attribute_columns_to_user.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        </update>
+
+        <addNotNullConstraint tableName="jhi_user" columnName="password_set_by_user" columnDataType="boolean"/>
+    </changeSet>
+
+    <!-- Add otp columns -->
+    <changeSet id="20241004170000_2" author="vosiievska">
+        <addColumn tableName="jhi_user">
+            <column name="otp_code" type="varchar(50)"/>
+            <column name="otp_code_creation_cate" type="timestamp"/>
```
Comment (sergeysenja1992) [inline src/main/resources/config/liquibase/changelog/20241004170000_add_otp_attribute_columns_to_user.xml:26]: 🐱 -> date

File: src/main/java/com/icthh/xm/uaa/domain/User.java [hunk 3]
```diff
@@ -67,6 +67,17 @@ public class User extends AbstractAuditingEntity implements Serializable {
...
 
+    @JsonIgnore
+    @NotNull
+    @Column(name = "password_set_by_user", nullable = false)
+    private Boolean passwordSetByUser; // wrapper should be used since there isn't default value
+
+    @Column(name = "otp_code", unique = true)
+    private String otpCode;
+
+    @Column(name = "otp_code_creation_cate")
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/uaa/domain/User.java:78]: 🐱 -> date

File: src/main/java/com/icthh/xm/uaa/service/dto/UserDTO.java (hunk 1 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/uaa/service/dto/UserDTO.java]: fixed

File: src/main/resources/config/liquibase/changelog/20241004170000_add_otp_attribute_columns_to_user.xml (hunk 2 above)
Comment (osiievska23) [reply src/main/resources/config/liquibase/changelog/20241004170000_add_otp_attribute_columns_to_user.xml]: fixed

File: src/main/java/com/icthh/xm/uaa/domain/User.java (hunk 3 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/uaa/domain/User.java]: fixed

**Pull request merged**